sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ConnectionConfig.java

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents the common connection configuration associated with a Cosmos Client in the Azure Cosmos DB database service.
+ */
+public abstract class ConnectionConfig {
+
+    protected final ConnectionMode connectionMode;
+
+    protected String userAgentSuffix;
+    protected  ThrottlingRetryOptions throttlingRetryOptions;
+    protected List<String> preferredRegions;
+    protected boolean endpointDiscoveryEnabled = true;
+    protected boolean usingMultipleWriteRegions = true;
+    protected Boolean readRequestsFallbackEnabled;
+
+    /**
+     * Constructor
+     * @param connectionMode connection mode
+     */
+    protected ConnectionConfig(ConnectionMode connectionMode) {
+        this.connectionMode = connectionMode;
+        this.throttlingRetryOptions = new ThrottlingRetryOptions();
+        this.readRequestsFallbackEnabled = null;
+        this.userAgentSuffix = "";
+    }
+
+    /**
+     * Gets the connection mode used in the client.
+     *
+     * @return the connection mode.
+     */
+    public ConnectionMode getConnectionMode() {
```

Comment:
One option is to inline these into the builder.

Reply:
Done.